### DISCUSSION TITLE
MAC Protocols for Wireless Sensor Networks - Showcase

### DISCUSSION THREAD

**Discussion by adamgeorge309** [inet-framework/inet#605] [Showcases]
Discuss the "MAC Protocols for Wireless Sensor Networks" showcase here. Ideas and recommendations are welcome.

**Comment by amirinia** [comment]
I ran showcase (omnetpp-5.5\samples\inet\showcases\wireless\sensornetwork) and I got this error
"unkonw directive: d -- in moudule(int:visulizer::infoCanvasVisualizer)", is there any solution?

**Comment by adamgeorge309** [comment]
Replace %d with %t in this line in omnetpp.ini:
`*.visualizer.*.infoVisualizer.format = "%d"`

Thanks for reporting this.

**Reply by OuhammouMourad** [reply]
Please hwo can i adopte this result on evaluating qos on WSN

**Reply by OuhammouMourad** [reply]
How can i use EDCA protocole

**Comment by aminawh** [comment]
i ran showcase (omnet-5.6\samples\inet\showcases\wireless\sensornetwork) and I got this error :
Module not found on path '^.networkNodeVisualizer' defined by par 'SensorNetworkShowcaseA.visualizer.canvasVisualizer.networkNodeVisualizerModule' -- in module (inet::visualizer::InfoCanvasVisualizer) SensorNetworkShowcaseA.visualizer.canvasVisualizer (id=13), during network initialization

what should i do please ?

**Comment by matteo463** [comment]
I ran showcase (omnetpp-5.5\samples\inet\showcases\wireless\sensornetwork) with and only with LMAC configuration sensornodes send only control signals: once ended the transmission of control it says "nothing else scheduled, get back to sleep".

What sould be the problem?

**Comment by abdulafroz04** [comment]
how can i able to get the graph of number of packets received vs slot duration for BMAC,LMAC,XMAC protcols.
and how to calculate total packets received, network power consumption,power consumption per packet.
please provide the procedure

**Reply by adamgeorge309** [reply]
You can have a look at the charts scripts in the showcase's .anf file (`Sensornetwork.anf`). These charts use the `best_performance.py` script in the showcase's folder, which contains functions to get data for the best performing simulation runs. The charts display the statistics you mentioned.

**Reply by abdulafroz04** [reply]
I have tried to plot the data using the Sensornetwork.anf, But couldn't able to get exact output. Can you help me out with this. I need plots of "number of packets received vs slot duration" as a Line chart and, "Total number of packets received", "Network total power consumption", "power consumption per packet " as bar chart. I need steps and what are to be selected for that graphs. Please Help me out with this.
![Screenshot (22)](https://github.com/inet-framework/inet/assets/109869836/4aac4128-6b7a-4b1e-9d84-36fd73b54c85)

**Comment by abdulafroz04** [comment]
can you provide me a detailed procedure of how to obtain the metrics and the graph for the BMAC, LMAC, XMAC protocols us. I couldn't able to get the data from the previous response

**Comment by rnapoli00** [comment]
Hi,
I'm trying to implement some PCAP writers on the BMac simulation. 
Reading the documentation, I included the following code:
```

*.sensor1.numPcapRecorders = 1
*.sensor1.pcapRecorder[*].pcapFile = "results/sensor1.pcap"
*.sensor1.pcapRecorder[*].moduleNamePatterns = "wlan[*]"

*.sensor2.numPcapRecorders = 1
*.sensor2.pcapRecorder[*].pcapFile = "results/sensor2.pcap"
*.sensor2.pcapRecorder[*].moduleNamePatterns = "wlan[*]"

*.sensor3.numPcapRecorders = 1
*.sensor3.pcapRecorder[*].pcapFile = "results/sensor3.pcap"
*.sensor3.pcapRecorder[*].moduleNamePatterns = "wlan[*]"

*.sensor4.numPcapRecorders = 1
*.sensor4.pcapRecorder[*].pcapFile = "results/sensor4.pcap"
*.sensor4.pcapRecorder[*].moduleNamePatterns = "wlan[*]"

*.server.numPcapRecorders = 1
*.server.pcapRecorder[*].pcapFile = "results/server.pcap"

**.pcapRecorder[*].alwaysFlush = true
```

Running the simulation I noticed that while for the gateway the udp packets are present on the PCAP file, the BMac preambles that the nodes send to agree on communication are not recorded.
How can I include them?
I apologize if this sounds like a stupid question, I am completely new to using INET.

**Reply by adamgeorge309** [reply]
I think BMac is not a supported link layer type in pcap

https://www.tcpdump.org/linktypes.html
